File: Code/CapstoneGroup2/Desktop Client/Model/Source.cs
using System;

namespace Desktop_Client.Model
{
    public enum SourceType
    {
        #region Enum members

        Pdf = 1,
        Vid = 2

        #endregion
    }

    public class Source
    {
        public int SourceId { get; set; }

        public string Type { get; set; }

        public SourceType NoteType => (SourceType)Enum.Parse(typeof(NoteType), this.Type, true);

        public string Name { get; set; }

        public bool IsLink { get; set; }

        public string Link { get; set; }
    }
}

File: Code/CapstoneGroup2/capstonegroup2.client/src/interfaces/Source.ts
// Source.ts
export enum SourceType {
    Pdf = 1,
    Vid = 2,
}

export const getSourceTypeName = (type: SourceType): string => {
    return type === SourceType.Pdf ? 'Pdf' : 'Vid';
};

export type Source = {
    sourceId: number;
    username: string;
    type: string;
    noteType: SourceType;
    name: string;
    description: string;
    isLink: boolean;
    link: string | null;
    content: string | null;
    createdAt: Date | null;
    updatedAt: Date | null;
    authorsString: string;
    authors: string[];
    publisher: string;
    accessedAt: Date | null;
};

